### PR TITLE
Comparison with Float (issue #79)

### LINKED ISSUE #79: exact comparisons with binary values/directed conversions
Currently
```julia
julia> using DecFP

julia> d64"1e100" == 1e100
true
```
which might seem reasonable, except that
```julia
julia> big(1e100)
1.000000000000000015902891109759918046836080856394528138978132755774783877217038e+100
```

Since the library doesn't appear to offer comparisons, the one option would be to do it via directed conversions, e.g.
```julia
<(x::Float64, y::Dec64) = x < Float64(y, RoundUp)
<=(x::Float64, y::Dec64) = x <= Float64(y, RoundDown)
==(x::Float64, y::Dec64) = x == Float64(y, RoundUp) == Float64(y, RoundDown)
```

Is it possible to do this without touching the global rounding mode?

### PR DESCRIPTION
**Pull request by jmkuhn**
Fixes #79

### CODE REVIEW

File: src/DecFP.jl [hunk 1]
```diff
@@ -612,6 +612,17 @@ function Base.:(==)(dec::DecimalFloatingPoint, rat::Rational)
...
 
+@eval Base.:(==)(dec::T, flt::Union{Float16,Float32,Float64}) where {T<:DecimalFloatingPoint} = dec == T(flt, RoundUp) == T(flt, RoundDown)
+@eval Base.:>(dec::T, flt::Union{Float16,Float32,Float64}) where {T<:DecimalFloatingPoint} = dec > T(flt, RoundDown)
+@eval Base.:<(dec::T, flt::Union{Float16,Float32,Float64}) where {T<:DecimalFloatingPoint} = dec < T(flt, RoundUp)
+@eval Base.:(>=)(dec::T, flt::Union{Float16,Float32,Float64}) where {T<:DecimalFloatingPoint} = dec > T(flt, RoundDown) || dec == T(flt, RoundUp)
+@eval Base.:(<=)(dec::T, flt::Union{Float16,Float32,Float64}) where {T<:DecimalFloatingPoint} = dec < T(flt, RoundUp) || dec == T(flt, RoundDown)
+@eval Base.:(==)(flt::Union{Float16,Float32,Float64}, dec::T) where {T<:DecimalFloatingPoint} = T(flt, RoundUp) == T(flt, RoundDown) == dec
+@eval Base.:>(flt::Union{Float16,Float32,Float64}, dec::T) where {T<:DecimalFloatingPoint} = T(flt, RoundUp) > dec
+@eval Base.:<(flt::Union{Float16,Float32,Float64}, dec::T) where {T<:DecimalFloatingPoint} = T(flt, RoundDown) < dec
+@eval Base.:(>=)(flt::Union{Float16,Float32,Float64}, dec::T) where {T<:DecimalFloatingPoint} = T(flt, RoundUp) > dec || T(flt, RoundDown) == dec
```
Comment (stevengj) [inline src/DecFP.jl:623]: Wouldn't `T(flt, RoundDown) >= dec` suffice?

**Comment by stevengj** [comment]
Very nice, this seems like a good way to do it!

File: src/DecFP.jl (hunk 1 above)
Comment (jmkuhn) [reply src/DecFP.jl]: Yes!  I've also eliminated the unnecessary `@eval`s leftover from an earlier version.

File: src/DecFP.jl [hunk 2]
```diff
@@ -612,6 +612,17 @@ function Base.:(==)(dec::DecimalFloatingPoint, rat::Rational)
     end
 end
 
+Base.:(==)(dec::T, flt::Union{Float16,Float32,Float64}) where {T<:DecimalFloatingPoint} = dec == T(flt, RoundUp) == T(flt, RoundDown)
+Base.:>(dec::T, flt::Union{Float16,Float32,Float64}) where {T<:DecimalFloatingPoint} = dec > T(flt, RoundDown)
+Base.:<(dec::T, flt::Union{Float16,Float32,Float64}) where {T<:DecimalFloatingPoint} = dec < T(flt, RoundUp)
+Base.:(>=)(dec::T, flt::Union{Float16,Float32,Float64}) where {T<:DecimalFloatingPoint} = dec >= T(flt, RoundUp)
+Base.:(<=)(dec::T, flt::Union{Float16,Float32,Float64}) where {T<:DecimalFloatingPoint} = dec <= T(flt, RoundDown)
+Base.:(==)(flt::Union{Float16,Float32,Float64}, dec::T) where {T<:DecimalFloatingPoint} = T(flt, RoundUp) == T(flt, RoundDown) == dec
```
Comment (stevengj) [inline src/DecFP.jl:620]: Would be clearer to define this as `dec == flt` since it is the same definition.

File: src/DecFP.jl [hunk 3]
```diff
@@ -612,6 +612,17 @@ function Base.:(==)(dec::DecimalFloatingPoint, rat::Rational)
...
+Base.:>(flt::Union{Float16,Float32,Float64}, dec::T) where {T<:DecimalFloatingPoint} = T(flt, RoundUp) > dec
+Base.:<(flt::Union{Float16,Float32,Float64}, dec::T) where {T<:DecimalFloatingPoint} = T(flt, RoundDown) < dec
+Base.:(>=)(flt::Union{Float16,Float32,Float64}, dec::T) where {T<:DecimalFloatingPoint} = T(flt, RoundDown) >= dec
+Base.:(<=)(flt::Union{Float16,Float32,Float64}, dec::T) where {T<:DecimalFloatingPoint} = T(flt, RoundUp) <= dec
```
Comment (stevengj) [inline src/DecFP.jl:624]: Can't you also define these as `>(flt, dec) = dec < flt` etcetera, to re-use the definitions from above?

**Pull request merged**